### PR TITLE
[VEN-2683]: deployment of XVSVault on zksyncsepolia

### PR DESCRIPTION
**Pull request by GitGuru7**
## Description

Resolves VEN-2683

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "clean": "hardhat clean && hardhat clean --config hardhat.config.zksync.ts"
   },
   "dependencies": {
+    "@defi-wonderland/smock": "2.3.5",
```
Comment (chechu) [inline package.json:34]: AFAIK, by including this in the `dependencies` section will increase the size of the public npm package. Do we need these dependencies apart from the development phases?

File: deploy/008-deploy-vaults.ts [hunk 2]
```diff
@@ -12,20 +12,26 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
     args: [],
     log: true,
     autoMine: true,
+    skipIfAlreadyDeployed: true,
+    maxFeePerGas: "200000000",
```
Comment (chechu) [inline deploy/008-deploy-vaults.ts:16]: Should this value depends on the network, as we made in other places?

File: tests/hardhat/Prime/Prime.ts [hunk 3]
```diff
@@ -190,7 +190,7 @@ async function deployProtocol(): Promise<SetupProtocolFixture> {
     ],
     {
       constructorArgs: [false, 10512000],
-      unsafeAllow: "constructor",
+      unsafeAllow: ["constructor", "internal-function-storage"],
```
Comment (chechu) [inline tests/hardhat/Prime/Prime.ts:193]: Why are we adding this? Could we add a comment where we are adding this flag, or at least a commit message with a body explaining the reason?

Comment (GitGuru7) [reply tests/hardhat/Prime/Prime.ts:194]: [51c0702](https://github.com/VenusProtocol/venus-protocol/pull/506/commits/51c070278168a0a65789c3b9e6320a1f57d5876c)

File: deploy/008-deploy-vaults.ts (hunk 2 above)
Comment (GitGuru7) [reply deploy/008-deploy-vaults.ts]: [51c0702](https://github.com/VenusProtocol/venus-protocol/pull/506/commits/51c070278168a0a65789c3b9e6320a1f57d5876c)

File: package.json (hunk 1 above)
Comment (GitGuru7) [reply package.json]: [51c0702](https://github.com/VenusProtocol/venus-protocol/pull/506/commits/51c070278168a0a65789c3b9e6320a1f57d5876c)

Comment (chechu) [reply package.json]: What about the dependency with `"hardhat": "2.19.5"`?

Comment (GitGuru7) [reply package.json]: We need this to pass CI checks

Comment (chechu) [reply package.json]: https://github.com/VenusProtocol/venus-protocol/pull/506/commits/4bd4d30182e490554c7d132bda0853bc68ce8d89

**Review by chechu — APPROVED** [review]

**Pull request merged**